### ISSUE TITLE
False negative for variable assignment with a value that's the same as a bash keyword

### ISSUE THREAD

**Issue by jmcantrell** [koalaman/shellcheck#2313]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2209
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

var=set

echo "$var"
```

#### Here's what shellcheck currently says:
```
Line 3	SC2209: Use var=$(command) to assign output (or quote to assign string).
```
#### Here's what I wanted or expected to see:

Nothing.
I think this would happen when the value of the variable is a name special to bash. Also triggers on `type` and `command`.

**Comment by koalaman** [comment]
You're right, see [the wiki page](https://github.com/koalaman/shellcheck/wiki/SC2209). 

Unfriendly it's hard to automatically differentiate it from its equivalent true positive:

```
vars=set
echo "$vars"
```

This is one of several warnings that are intended to help absolute beginners, in this case those who aren't familiar with command substitution. Since you're clearly not in that category it's unlikely to ever be helpful. Please don't hesitate to banish this suggestion in your .shellcheckrc.

**Comment by jmcantrell** [comment]
That’s fair. I probably should have done that from the outset.
